tests/taker_cli.rs

```diff
@@ -0,0 +1,321 @@
+mod test {
+
+    use bitcoin::{address::NetworkChecked, Address, Amount, Network};
+    use bitcoind::{
+        bitcoincore_rpc::{Auth, RpcApi},
+        tempfile::tempdir,
+        BitcoinD, Conf,
+    };
+    use coinswap::{
+        maker::{start_maker_server, Maker, MakerBehavior},
+        taker::{ Taker, TakerBehavior},
+        utill::read_connection_network_string,
+        wallet::RPCConfig,
+    };
+    use std::{path::PathBuf, process::Command, str::FromStr, sync::Arc, thread};
+    /// Testing errors for integration tests
+    #[derive(Debug)]
+    enum IntTestError {
+        // IO error
+        IO(std::io::Error),
+        // Command execution error
+        CmdExec(String),
+        }
+
+    struct TakerCli {
+        target: String,
+        rpc_network: String,
+        network: String,
+        data_dir: Option<PathBuf>,
+        taker: Taker, 
+        makers: Vec<Arc<Maker>>,
+        bitcoind: BitcoinD,
+    }
+
+    impl TakerCli {
+        fn new(
+            data_dir: Option<PathBuf>,
+            rpc_network: &str,
+            network: &str,
+            makers_count: u16,
+            wallet_name: Option<String>,
+        ) -> TakerCli {
+            //---------------------- start bitcoind--------------------------------------------------------
+            // Initiate bitcoind instance
+
+            let temp_dir = tempdir().unwrap().into_path();
+
+            let mut conf = Conf::default();
+
+            conf.args.push("-txindex=1"); //txindex is must, or else wallet sync won't work.
+            conf.staticdir = Some(temp_dir.join(".bitcoin"));
+
+            let key = "BITCOIND_EXE";
+            let curr_dir_path = std::env::current_dir().unwrap();
+            let bitcoind_path = curr_dir_path.join("bin").join("bitcoind");
+            std::env::set_var(key, bitcoind_path);
+            let exe_path = bitcoind::exe_path().unwrap();
+
+            let bitcoind = BitcoinD::with_conf(exe_path, &conf).unwrap();
+
+            //--------------------Set Up for Maker & Taker---------------------------------------
+
+            // make rpc_config from bitcoind
+            let url = bitcoind.rpc_url().split_at(7).1.to_string();
+            let auth = Auth::CookieFile(bitcoind.params.cookie_file.clone());
+
+            let rpc_config = RPCConfig {
+                url,
+                auth,
+                network: Network::from_str(rpc_network).unwrap(),
+                wallet_name: wallet_name
+                    .clone()
+                    .unwrap_or_else(|| String::from("WALLET")),
+            };
+
+            let connection_type = read_connection_network_string(network).unwrap();
+
+            // create taker instance
+            let mut taker = Taker::init(
+                data_dir.clone(),
+                wallet_name,
+                Some(rpc_config.clone()),
+                TakerBehavior::Normal,
+                Some(connection_type),
+            )
+            .unwrap();
```

Comment:
Nope, we can't init taker this way. You need to call the `taker` command via std::process::Command with all the right args. And then collect the stdout response from the process. Assert value from that response.